src/anemoi/training/diagnostics/callbacks/__init__.py

```diff
@@ -721,6 +739,43 @@ def _plot(
                 exp_log_tag=f"val_pred_sample_rstep{rollout_step:02d}_rank{local_rank:01d}",
             )
 
+            # check if stretched grid
+            if "lam_resolution" in getattr(self.config.graph.nodes.hidden, "node_builder", []):
+                fig_lam_inside = plot_predicted_multilevel_flat_sample(
+                    plot_parameters_dict,
+                    self.config.diagnostics.plot.per_sample,
+                    self.latlons[pl_module.mask],
+                    self.config.diagnostics.plot.accumulation_levels_plot,
+                    self.config.diagnostics.plot.cmap_accumulation,
+                    data[0, :, pl_module.mask, :].squeeze(),
+                    data[rollout_step + 1, :, pl_module.mask, :].squeeze(),
+                    output_tensor[rollout_step, :, pl_module.mask, :],
+                )
+                self._output_figure(
+                    logger,
+                    fig_lam_inside,
+                    epoch=epoch,
+                    tag=f"pred_val_sample_inside_lam_rstep{rollout_step:02d}_batch{batch_idx:04d}_rank0",
+                    exp_log_tag=f"val_pred_sample_inside_lam_rstep{rollout_step:02d}_rank{local_rank:01d}",
+                )
+                fig_lam_outside = plot_predicted_multilevel_flat_sample(
+                    plot_parameters_dict,
+                    self.config.diagnostics.plot.per_sample,
+                    self.latlons[~pl_module.mask],
+                    self.config.diagnostics.plot.accumulation_levels_plot,
+                    self.config.diagnostics.plot.cmap_accumulation,
+                    data[0, :, ~pl_module.mask, :].squeeze(),
+                    data[rollout_step + 1, :, ~pl_module.mask, :].squeeze(),
+                    output_tensor[rollout_step, :, ~pl_module.mask, :],
+                )
+                self._output_figure(
+                    logger,
+                    fig_lam_outside,
+                    epoch=epoch,
+                    tag=f"pred_val_sample_outside_lam_rstep{rollout_step:02d}_batch{batch_idx:04d}_rank0",
+                    exp_log_tag=f"val_pred_sample_outside_lam_rstep{rollout_step:02d}_rank{local_rank:01d}",
+                )
+
```

Comment:
It may be good to have these plotting calls in another callback specific to StretchedGrid, that way a user can choose to toggle on or off. Then an assert can be raised when attempting to plot without the appropriate config.